### PR TITLE
Support quant_method of modelopt_mixed

### LINKED ISSUE #422: modelopt_mixed format for nvidia/Qwen3.5-397B-A17B-NVFP4-V2
`nvidia/Qwen3.5-397B-A17B-NVFP4-V2` uses `modelopt_mixed` which throws
```rust
vllm-rs-svc0  | 2026-07-06T17:42:04.494146Z  WARN xinfer::utils: Multimodel model Some(["Qwen3_5MoeForConditionalGeneration"]) detected!
vllm-rs-svc0  | 
vllm-rs-svc0  | thread 'main' (1) panicked at src/utils/mod.rs:1488:13:
vllm-rs-svc0  | Invalid quantization format! Only `gptq`, `awq`, `fp8`, `mxfp4` and `nvfp4` supported, got `modelopt_mixed`
vllm-rs-svc0  | stack backtrace:
vllm-rs-svc0  |    0:     0x55e35da2109a - <<std[52919eca6bce4da3]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[18c8dd30382e7099]::fmt::Display>::fmt
vllm-rs-svc0  |    1:     0x55e35da3daaa - core[18c8dd30382e7099]::fmt::write
vllm-rs-svc0  |    2:     0x55e35da292c2 - <std[52919eca6bce4da3]::sys::stdio::unix::Stderr as std[52919eca6bce4da3]::io::Write>::write_fmt
vllm-rs-svc0  |    3:     0x55e35d9fb11f - std[52919eca6bce4da3]::panicking::default_hook::{closure#0}
vllm-rs-svc0  |    4:     0x55e35da17d71 - std[52919eca6bce4da3]::panicking::default_hook
vllm-rs-svc0  |    5:     0x55e35da17feb - std[52919eca6bce4da3]::panicking::panic_with_hook
vllm-rs-svc0  |    6:     0x55e35d9fb1d8 - std[52919eca6bce4da3]::panicking::panic_handler::{closure#0}
vllm-rs-svc0  |    7:     0x55e35d9efa09 - std[52919eca6bce4da3]::sys::backtrace::__rust_end_short_backtrace::<std[52919eca6bce4da3]::panicking::panic_handler::{closure#0}, !>
vllm-rs-svc0  |    8:     0x55e35d9fc39d - __rustc[8068f81614cfe5c]::rust_begin_unwind
vllm-rs-svc0  |    9:     0x55e35da3e24c - core[18c8dd30382e7099]::panicking::panic_fmt
vllm-rs-svc0  |   10:     0x55e35c6f889b - xinfer::utils::init_config_tokenizer::hbfc8ee8be78340a8
vllm-rs-svc0  |   11:     0x55e35c588c64 - xinfer::core::engine::LLMEngine::new::had43708ecd0c2032
vllm-rs-svc0  |   12:     0x55e35c3de7b0 - xinfer::main::{{closure}}::h679167590597560d
vllm-rs-svc0  |   13:     0x55e35c3db04c - xinfer::main::h3b3fb253e4adf253
vllm-rs-svc0  |   14:     0x55e35c44cd06 - std::sys::backtrace::__rust_begin_short_backtrace::hcbb0e8bf334efb52
vllm-rs-svc0  |   15:     0x55e35c42d535 - std::rt::lang_start::{{closure}}::h46a95ecc2bdbdc0a
vllm-rs-svc0  |   16:     0x55e35da16894 - std[52919eca6bce4da3]::rt::lang_start_internal
vllm-rs-svc0  |   17:     0x55e35c3eef55 - main
vllm-rs-svc0  |   18:     0x7f6d5bdbdd90 - <unknown>
vllm-rs-svc0  |   19:     0x7f6d5bdbde40 - __libc_start_main
vllm-rs-svc0  |   20:     0x55e35c2e43a5 - _start
vllm-rs-svc0  |   21:                0x0 - <unknown>
```
on current `main`

### PR DESCRIPTION
**Pull request by guoqingbao**
Resolve #422

### CODE REVIEW

**Comment by sempervictus** [comment]
`nvidia/Qwen3.5-397B-A17B-NVFP4-V2` loads! It also responds with _words_ but their applicability is questionable:

image

This is what the older `nvidia/Qwen3.5-397B-A17B-NVFP4` used to eventually do as well... but not at **the first turn** of a conversation. Sometimes it takes 3-4 turns but the longer it takes the more messed up the problem appears:

`aichat -m g60 --role %research% "search for the xinfer rust runtime and explain what it is"`

> \<think>
> 
> The user is asking me to search for "xinfer rust runtime" and explain what it is. I need to use my web search to find information about this topic. Let me search for information about xinfer and its Rust runtime.
> 
> I'll start by searching for information about xinfer to understand what it is, then look search for relevant information about its Rust runtime component.
> 
> \</think>
> 
> 
> 
> Call web_search_searxng {"query":"xinfer rust runtime"}
> Processing 4096 tokens per chunk
> \<think>
> 
> ...
> 
> \</think>
> 
> 
> 
> Call fetch_url_via_curl {"url":"https://githubhttps://guoqingbao.github.com"}
> curl: (6) Could not resolve host: githubhttps
> \<think>
> 
> ...
> 
> \</think>
> 
> 
> 
> Call fetch_url_via_curl {"url":"httpshttps://gu.guoqingbaoao/xinfer/rust"}
> curl: (1) Protocol "httpshttps" not supported
> \<think>
> 
> ...
> ",\"category":0.11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111Error: Aborted.
> 

^^ above happens w/ 2-phase, no grammar at all, or full-envelope masking. I dont think its anything **we** are doing to the logits given that **no** grammar has identical effect to _full_ grammar - the collapse occurs somewhere before we get logprobs because the emitted text can "bypass" explicit grammar-mandated tokens meaning they're not in the set of logits available at all to be sampled exclusively (since that mask is effectively ^"everything but this specific token").

I was assuming they released `nvidia/Qwen3.5-397B-A17B-NVFP4-V2` because there was some structural problem with V1 but now i'm wondering if whatever is causing NVFP4 models in `xinfer` to misbehave is actually being amplified by the variable quantization or possibly by how the runtime is loading/accessing/executing over the quantized or dequantized values.

**Comment by guoqingbao** [comment]
> `nvidia/Qwen3.5-397B-A17B-NVFP4-V2` loads! It also responds with _words_ but their applicability is questionable:

Have you made this working with current main? it needs per-tensor weight scale support.

**Comment by sempervictus** [comment]
Yes, made sure `main`  was the base and this PR was the variable. After getting those results i tried adding #421 and #358 just in case they might help offset the adverse effect.

The 80B FP4 is rather stable, the 122B can get weird at ~3-4X context length, but the 397B goes mad at fractions of its originally trained max embedding positions. Other large models like llama and minimax2 suffer degeneration as well but it manifests differently - mm2 has some sort of attention problem where it will "skip sections of input" entirely omitting it from consideration and the llama models just output patent nosense. Given the behavioral difference i dont think that they share a cause but it does strike me as odd that the stability threshold of FP4 models seems to be around 100b when the quantization itself is intended to allow larger models to be used in ~cconsumer kit.

**Comment by guoqingbao** [comment]
> Yes, made sure `main` was the base and this PR was the variable. After getting those results i tried adding #421 and #358 just in case they might help offset the adverse effect.

This model, nvidia/Qwen3.5-397B-A17B-NVFP4-V2, uses their older quantization method. I'm not sure whether the latest form requires a CUTLASS update, but their older NVFP4 models themselves have precision issues for sure.
